src/library/scala/collection/convert/StreamExtensions.scala

```diff
@@ -20,11 +20,15 @@ import scala.annotation.implicitNotFound
 import scala.collection.AnyConstr
 import scala.collection.convert.impl.{AccumulatorFactoryInfo, StepperShape, StreamShape, StreamUnboxer}
 
-trait StreamConverters {
+trait StreamExtensions {
   // collections
 
   implicit class IterableHasSeqStream[A](cc: IterableOnce[A]) {
-    def seqStream[S <: BaseStream[_, S], St <: Stepper[_]](implicit s: StreamShape[A, S, St], st: StepperShape[A, St]): S =
+    /** Create a sequential [[java.util.stream.Stream Java Stream]] for this collection. If the
+      * collection contains primitive values, a corresponding specialized Stream is returned (e.g.,
+      * [[java.util.stream.IntStream `IntStream`]]).
+      */
+    def asJavaSeqStream[S <: BaseStream[_, S], St <: Stepper[_]](implicit s: StreamShape[A, S, St], st: StepperShape[A, St]): S =
```

Comment:
Oh ugh.  If we want to scare people away from using these, extra-long names is a good way to do it.  Is there any way to make this more ergonomic?  I agree with the desire to clearly mark them, but it still feels clumsy, especially when you have to add `Key` or `Value`

Reply:
OK, thanks for the clear vote :-) I added `asJava` to keep all the conversion methods in `scala.jdk.FooConverters` uniform. Now they are all of the form `[as|to][Java|Scala](Target)?`
  - `as` for views (like the conversions to classic Java collections), `to` for actual conversions where the result doesn't have any reference to the original (like `option.toJava`)
  - a `Target` is there for conversions to a less common type (for example `map.asJava: java.util.Map` but `map.asJavaDictionary: java.util.Dictionary`), or to dis-ambiguate when there's no clear default (`c.asJavaSeqStream` vs `c.asJavaParStream`)

We could make an exception for `StreamConverters` and drop the `asJava` / `asScala`. I'm not against.

Reply:
`asJava` is used when you know what the Java variant is; the Scala type determines the Java one (and vice versa).  This is different; you're calling a method to generate something that happens to be in the Java library, but for which there's no Scala equivalent.  The method name states what you're after.  So I think it would be okay to make an "exception", or rather, have a rule `asJava` and `toJava` are only used when the target type is inferred or when there are both Scala and Java types of the same function and name.  (Since Scala's `Stream` is now deprecated, I wouldn't count it here.)